SQLiteStudio3/Tests/TestUtils/collationmanagermock.cpp

```diff
@@ -13,6 +13,11 @@ QList<CollationManager::CollationPtr> CollationManagerMock::getAllCollations() c
     return QList<CollationManager::CollationPtr>();
 }
 
+CollationManager::CollationPtr CollationManagerMock::getCollation(const QString&) const
+{
+   return nullptr;
```

Comment:
Is this allowed to return `nullptr` here? I mean the `CollationPtr`, as far as I remember, is alias for `QSharedPointer<Collation>`, in which case this will be a syntax error, wouldn't it? Or is this handled by some overloaded operator, making `nullptr` implicitly casted to any `QSharedPointer` that is needed?

Reply:
Well, I hadn't considered that yet. My code is returning `nullptr` from`CollationManager::getCollation()` in the case of unknown collation and handles this case in the only place it's used. I am also not getting any syntax errors...

Reply:
You're right. Such case is already there. Interesting to learn it.